### PR TITLE
Switch to default Travis + Codecov integration

### PR DESCRIPTION
**Pull request by fingolfin**
This packages is one of the few *not* using the default set of scripts, and I figured it might make it easier to debug these things to use the "standard" solution.

... assuming this even works, we'll see ;-).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,47 +1,36 @@
 language: cpp
-sudo: false
+env:
+  global:
+    - GAPROOT=gaproot
+    - COVDIR=coverage
+    - GAP_PKGS_TO_BUILD="io" # exclude profiling, i.e., *this* package
+    - NO_COVERAGE=1  # tests are not compatible with regular coverage collection
```
Comment (fingolfin) [inline .travis.yml]: Had to disable coverage computation again as the test suite sadly does not like that, it expects to start with profiling disabled. While that is understandable, it is also a bit sad, as the coverage results would of course be useful. I guess one could solve this with some cleverness, but that's beyond the scope of this PR.

File: .travis.yml [hunk 2]
```diff
@@ -1,47 +1,36 @@
...
-         - libgmp-dev
-
+    - env: GAPBRANCH=master ABI=32 PKGOPTS="CFLAGS=-m32 LDFLAGS=-m32 LOPTS=-m32 CXXFLAGS=-m32"
+    - env: GAPBRANCH=master
+    - env: GAPBRANCH=stable-4.10
 
-install:
-  - git clone --depth=2 https://github.com/gap-system/gap.git gapsrc -b ${BRANCH}
-  - pwd
-  - ( cd gapsrc && (./autogen.sh || true) && ./configure --with-gmp=system && make configure && make -j2 && make bootstrap-pkg-minimal)
```
Comment (fingolfin) [inline .travis.yml]: So one major drawback of this PR is that it switches from `bootstrap-pkg-minimal` to `bootstrap-pkg-full` which is of course slower, as more data has to be downloaded.

But I wouldn't mind adding an option (e.g. via an env var) to the [pkg-ci-scripts](https://github.com/gap-system/pkg-ci-scripts.git) to allow changing this from full to minimal for authors who want that...

File: .travis.yml (hunk 1 above)
Comment (ChrisJefferson) [reply .travis.yml]: It's hard to imagine how to easily test profiling profiling :) could write a very different test suite which first generates some profiles, then test reading and outputting them (which could be profiled).

File: .travis.yml (hunk 2 above)
Comment (fingolfin) [reply .travis.yml]: I've implemented this now, via GAP_BOOTSTRAP=minimal

**Comment by fingolfin** [comment]
This seems to work pretty well now.

**Comment by ChrisJefferson** [comment]
will merge and adjust

**Pull request merged**